File: fixtures/scale/java/large/src/cryptobench/ciphers/playfair/PlayfairCipher.java
package cryptobench.ciphers.playfair;

import cryptobench.core.Cipher;
import cryptobench.util.Alphabet;

/** Digraph substitution over a 5x5 keyword square. */
public class PlayfairCipher implements Cipher {
    private final String square;

    public PlayfairCipher(PlayfairKey key) {
        this.square = key.square();
    }

    @Override
    public String name() {
        return "playfair";
    }

    @Override
    public String encrypt(String plaintext) {
        return PlayfairRules.transform(square, PlayfairDigraphs.split(Alphabet.clean(plaintext)), 1);
    }

    @Override
    public String decrypt(String ciphertext) {
        return PlayfairRules.transform(square, PlayfairDigraphs.split(Alphabet.clean(ciphertext)), 4);
    }
}

File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/playfair/PlayfairCipher.kt
package cryptobench.ciphers.playfair

import cryptobench.core.Cipher
import cryptobench.util.Alphabet

/** Digraph substitution over a 5x5 keyword square. */
class PlayfairCipher(key: PlayfairKey) : Cipher {
    private val square = key.square()

    override fun name(): String = "playfair"

    override fun encrypt(plaintext: String): String =
        PlayfairRules.transform(square, PlayfairDigraphs.split(Alphabet.clean(plaintext)), 1)

    override fun decrypt(ciphertext: String): String =
        PlayfairRules.transform(square, PlayfairDigraphs.split(Alphabet.clean(ciphertext)), 4)
}
